### PR TITLE
Added handling of space in -I/-isystem switch if path is separated from it

### PR DESCRIPTION
**Pull request by xvoidee**

### CODE REVIEW

**Comment by xvoidee** [comment]
If compilation database contains something like this:
-isystem bla -I bla
Then chromatica not able to parse it and will not place switch and path in a single row in ChromaticaShowInfo output.

File: rplugin/python3/chromatica/compile_args_database.py [hunk 1]
```diff
@@ -190,7 +191,13 @@ def get_cdb_args(self, filename):
                     if _basename in arg:
                         continue;
                     else:
-                        res.append(arg)
+                        # if last added switch was standalone include then we need to append path to it
+                        if last == '-I' or last == '-isystem':
+                            res[len(res) - 1] += arg
```
Comment (pierremoreau) [inline rplugin/python3/chromatica/compile_args_database.py]: I think you need to apply
```
if not os.path.isabs(arg):
    arg = os.path.normpath(os.path.join(cwd, arg))
```
before appending it, as done in the cases where there are no spaces between `-I` or `-isystem`.

**Comment by pierremoreau** [comment]
I wonder whether the relative -> absolute path resolving should be turned into a function as the same code is now found in at least three different places (granted it’s only two lines of code).

By the way, thank you for submitting this MR: I have been running into that issue and only made some quick hack locally.

File: rplugin/python3/chromatica/compile_args_database.py (hunk 1 above)
Comment (pierremoreau) [reply rplugin/python3/chromatica/compile_args_database.py]: I saw it, thank you for addressing it, but be aware that I am only a user and contributor and can’t approve your merge request.

**Pull request merged**